airflow/providers/amazon/aws/operators/neptune.py

```diff
@@ -0,0 +1,222 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Sequence
+
+from airflow.configuration import conf
+from airflow.providers.amazon.aws.hooks.neptune import NeptuneHook
+from airflow.providers.amazon.aws.operators.base_aws import AwsBaseOperator
+from airflow.providers.amazon.aws.triggers.neptune import (
+    NeptuneClusterAvailableTrigger,
+    NeptuneClusterStoppedTrigger,
+)
+from airflow.providers.amazon.aws.utils.mixins import aws_template_fields
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+AVAILABLE_STATES = ["available"]
+STOPPED_STATES = ["stopped"]
```

Comment:
I think this is better to be defined on the hook level so it can be used by multiple operators/sensors.
Example:

https://github.com/apache/airflow/blob/fd03dc29336e1331d20de0113993dd5a35353ee0/airflow/providers/amazon/aws/hooks/athena.py#L55-L68

https://github.com/apache/airflow/blob/99f320354b075fb780e54057d223d2d16ddf08b8/airflow/providers/amazon/aws/hooks/batch_client.py#L190-L199